### PR TITLE
[GPU] Fix syntax error of fully_connected_gpu_fb_io_block_fp16 kernel

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Fix syntax error of fully_connected_gpu_fb_io_block_fp16 kernel
 - Add related func test

### Tickets:
 - 139622

### CODE REVIEW

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/mat_mul.cpp [hunk 1]
```diff
@@ -10,6 +10,27 @@ namespace {
...
+        // Some rounding float to integer types on GPU may differ from CPU, and as result,
+        // the actual values may differ from reference ones on 1 when the float is very close to an integer,
+        // e.g 6,0000023 calculated on CPU may be cast to 5 by OpenCL convert_uchar function.
+        // That is why the threshold is set 1.f for integer types.
+        if (targetDevice == "GPU" && model_type == ov::element::f16) {
+            rel_threshold = 0.005;
+        }
+    }
+};
+
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/mat_mul.cpp:33]: I think test should be implemented from unittest with enforcing kernel instance.

Comment (wilson-seok) [reply src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/mat_mul.cpp:33]: Added unit test for the purpose. Thanks!

File: src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp [hunk 2]
```diff
@@ -1081,6 +1081,67 @@ TEST(fully_connected_gpu, DISABLED_fs_byx_fsv32_b34)
...
+    // Allocate memory
+    auto input_mem = engine.allocate_memory({ { batch_num, in_feature_num }, data_types::f16, format::bfyx });
+    auto weights_mem = engine.allocate_memory({ { out_feature_num, in_feature_num }, data_types::f16, format::bfyx });
+
+    // Generate random input data and set values
+    auto input_data = rg.generate_random_1d<ov::float16>(batch_num * in_feature_num, -1, 1);
+    auto weights_data = rg.generate_random_1d<ov::float16>(out_feature_num * in_feature_num, -1, 1);
+
+    // auto input_data_bfyx = flatten_4d(format::bfyx, input_data);
+    // auto weights_data_bfyx = flatten_4d(format::bfyx, weights_data);
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp:1101]: nit: please remove unnecessary comment

Comment (wilson-seok) [reply src/plugins/intel_gpu/tests/unit/test_cases/fully_connected_gpu_test.cpp]: Updated! Thanks!

**Review by sshlyapn — APPROVED** [review]

File: src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/mat_mul.cpp [hunk 3]
```diff
@@ -10,6 +10,27 @@ namespace {
...
+        std::vector<ov::test::InputShape> shapes;            // Input Shapes
+        std::pair<bool, bool> transpose_type;              // Transpose inputs
+        ov::element::Type model_type;                  // Model type
+        ov::test::utils::InputLayerType input_layer_type;    // Secondary input type
+        std::map<std::string, std::string>  additional_config; // Additional network configuration
+        std::tie(shapes, transpose_type, model_type, input_layer_type, targetDevice, additional_config) = this->GetParam();
+        // Some rounding float to integer types on GPU may differ from CPU, and as result,
+        // the actual values may differ from reference ones on 1 when the float is very close to an integer,
+        // e.g 6,0000023 calculated on CPU may be cast to 5 by OpenCL convert_uchar function.
+        // That is why the threshold is set 1.f for integer types.
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/mat_mul.cpp:27]: The comment seems to be irrelevant

Comment (wilson-seok) [reply src/plugins/intel_gpu/tests/functional/shared_tests_instances/single_layer_tests/mat_mul.cpp:27]: oops! my fault! Will remove it in next change.
